apps/server/src/modules/files-storage/client/s3-client.adapter.ts

```diff
@@ -183,6 +186,61 @@ export class S3ClientAdapter implements IStorageClient {
 		return this.client.send(req);
 	}
 
+	public async list(prefix: string, maxKeys?: number) {
```

Comment:
Why did you remove the default value of Infinity for maxKeys? Did it lead to problems? Now the logic below is a bit uglier because you always have to check first if maxKeys is set.

Reply:
Ah okay. Thanks for explanation :)